src/vm/eventpipebuffermanager.cpp

```diff
@@ -202,6 +220,25 @@ EventPipeBufferManager::EventPipeBufferManager()
     m_numEventsDropped = 0;
     m_numEventsWritten = 0;
 #endif // _DEBUG
+
+    m_pCurrentEvent = nullptr;
+    m_pCurrentBuffer = nullptr;
+    m_pCurrentBufferList = nullptr;
+
+    m_maxSizeOfAllBuffers = MIN(MAX(100 * 1024, pSession->GetCircularBufferSize()), (ULONGLONG)10 * 1024 * 1024 * 1024);
+
+    if (pSession->GetSequencePointAllocationBudget() == 0)
```

Comment:
nit: `SequencePointAllocationBudget` could be passed as parameter, and you would not need to pass the `EventPipeSession` object.

Reply:
In later parts of the change I capture it as a member field and use it in other functions.